design/apidsl/media_type.go

```diff
@@ -355,6 +364,8 @@ func Link(name string, view ...string) {
 	}
 }
 
+// TODO: is this correct?
+// CollectionOf can be used in: Media, Response, ResponseTemplate
```

Comment:
for CollectionOf I think it is a good idea to say it can be used where a MediaType can be used, but then where can someone go to see all the possible places that a MediaType identifier can be used? Should we add a list to the MediaType docstring to include all the places a MediaType identifier can be used?